.github/workflows/java-pr.yml

```
@@ -31,6 +31,10 @@ on:
     - cron: "0 */12 * * *"
   workflow_dispatch:
 
+concurrency:
+  group: java-pr-${{ github.event.number }}
```

```suggestion
  group: java-pr-${{ github.event.issue.number || github.run_id }}
```

Could we do something like this instead? This will only impact pull request runs (will limit to one run per pr), but will not interfere with scheduled or manual runs (which I don't think we should cancel)

---

I think if we add a `run_id` here then this will not work as intended since multiple runs of the same PR will continue to run in parallel? 

`run_id` is effectively the thing we want to run "one at a time" I think.

---

`github.event.issue.number || github.run_id` will first check if there is an issue number, which will be true for all PRs. So PRs will not use `run_id`

---

See https://docs.github.com/en/actions/using-jobs/using-concurrency#example-using-a-fallback-value

---

Perfect, thanks for the ref. Making the change!